src/component/component_system_manager.cc

```diff
@@ -131,24 +131,49 @@ set_door_state(ship_space *ship, c_entity ce, surface_type s)
     auto position = pos_man.get_instance_data(ce);
     auto pos = glm::ivec3(*position.position);
     auto door = door_man.get_instance_data(ce);
+    auto rot = *position.rotation;
 
     auto from_surface = s == surface_none ? surface_door : surface_none;
 
-    /* todo: this has no support for rotation whatsoever */
     for (auto h = 0; h < *door.height; ++h) {
-        auto ym = glm::ivec3(pos.x, pos.y - 1, pos.z);
-        auto yp = glm::ivec3(pos.x, pos.y + 1, pos.z);
+        
+        float x_change = glm::round(glm::sin(rot));
+        float y_change = glm::round(glm::cos(rot));
+
+        surface_index surf_plus;
+        surface_index surf_minus;
+
+        if (x_change > 0) {
+            surf_plus = surface_xp;
```

Comment:
There's a helper that will give you these, too.

Note the opposites differ only by their low bit, too.